pkg/telemetry/prometheus_resourcer_test.go

```diff
@@ -34,11 +34,11 @@ func Test_PrometheusResourcer_UpdateResources_Create_CassDC(t *testing.T) {
 		CommonLabels:         map[string]string{k8ssandraapi.K8ssandraClusterNameLabel: "test-k8ssandracluster"},
 	}
 	ownerCassDC := test.NewCassandraDatacenter("test-cassdc", "test-namespace")
-	serviceMonitor, err := cfg.NewCassServiceMonitor()
+	serviceMonitor, err := cfg.NewCassServiceMonitor(true)
 	if err != nil {
 		assert.Fail(t, "couldn't create new ServiceMonitor for CassDC", "error", err)
 	}
-	if cfg.UpdateResources(ctx, fakeClient, &ownerCassDC, &serviceMonitor); err != nil {
+	if err := cfg.UpdateResources(ctx, fakeClient, &ownerCassDC, serviceMonitor); err != nil {
```

Comment:
Blatant error that the compiler happily compiles...

Reply:
😅 
The compiler is very kind to us.